### PR TITLE
Deprecated NSG Flow Log Policies

### PR DESCRIPTION
**Pull request by JamJarchitect**
## Overview/Summary

Deprecation of two ALZ Policies:
- Deploy-Nsg-FlowLogs.json
- Deploy-Nsg-FlowLogs-to-LA.json

There is a built-in policy which covers the functionality of those two policies to storage accounts and log analytics.

## This PR fixes/adds/changes/removes

1. Deprecates ALZ custom policy Deploy-Nsg-FlowLogs.json
2. Deprecates ALZ custom policy Deploy-Nsg-FlowLogs-to-LA.json

## Testing Evidence

Deploy-Nsg-FlowLogs.json:
https://www.azadvertizer.net/azpolicyadvertizer/Deploy-Nsg-FlowLogs.html

Deploy-Nsg-FlowLogs-to-LA.json:
https://www.azadvertizer.net/azpolicyadvertizer/Deploy-Nsg-FlowLogs-to-LA.html

The built-in policy in question:
https://www.azadvertizer.net/azpolicyadvertizer/e920df7f-9a64-4066-9b58-52684c02a091.html

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fnsgflowdeprecated%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fnsgflowdeprecated%2FeslzArm%2Feslz-portal.json)

#### Azure US Gov (Fairfax)
[Deploy To Azure](https://portal.azure.us/#blade/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fnsgflowdeprecated%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FJamJarchitect%2FEnterprise-Scale%2Fnsgflowdeprecated%2FeslzArm%2Ffairfaxeslz-portal.json)

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Hey @JamJarchitect can we also update the what's new wiki page with these changes. And I'm here can we suggest the policy that has replaced this as a built in in a table?

Like this
Old policy ID | new policy ID 
Xxxxxxxxxxxxx | xxxxxxxxxxxxx

Thanks

**Comment by JamJarchitect** [comment]
> https://www.azadvertizer.net/azpolicyadvertizer/e920df7f-9a64-4066-9b58-52684c02a091.html

@jtracey93 :
- Updated the What's New
- Created the table under November 2022

File: docs/wiki/Whats-new.md [hunk 1]
```diff
@@ -2,22 +2,86 @@
 
 - [In this Section](#in-this-section)
 - [Updates](#updates)
+  - [November 2022](#november-2022)
+    - [Policy](#policy)
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:6]: Looks like new subheading have been dragged in by mistake. I suspect an extension has done this in VSCode

File: docs/wiki/Whats-new.md [hunk 2]
```diff
@@ -46,6 +110,17 @@ This article will be updated as and when changes are made to the above and anyth
 
 Here's what's changed in Enterprise Scale/Azure Landing Zones:
 
+### November 2022
+
+#### Policy
+
+- Added ALZ Azure Policy [deprecation process](Deprecating-ALZ-Policies.md) to the Wiki.
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:117]: Move this to the docs subheading in Nov 2022

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: docs/wiki/Whats-new.md (hunk 2 above)
Comment (JamJarchitect) [reply docs/wiki/Whats-new.md:117]: Sorted out subheadings

File: docs/wiki/Whats-new.md (hunk 1 above)
Comment (JamJarchitect) [reply docs/wiki/Whats-new.md]: Just months are in the table now

File: docs/wiki/Whats-new.md [hunk 3]
```diff
@@ -2,7 +2,6 @@
 
 - [In this Section](#in-this-section)
 - [Updates](#updates)
-  - [November 2022](#november-2022)
   - [October 2022](#october-2022)
```
Comment (jtracey93) [inline docs/wiki/Whats-new.md:5]: Missing the Nov 2022 TOC item @JamJarchitect

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: docs/wiki/Whats-new.md (hunk 3 above)
Comment (JamJarchitect) [reply docs/wiki/Whats-new.md]: 🤦‍♂️ Added it now. Thanks @jtracey93

File: src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Nsg-FlowLogs.json [hunk 4]
```diff
@@ -1,15 +1,16 @@
 {
-  "name": "Deploy-Nsg-FlowLogs",
+  "name": "[Deprecated] Deploy-Nsg-FlowLogs",
```
Comment (jtracey93) [inline src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Nsg-FlowLogs.json:2]:
```suggestion
  "name": "Deploy-Nsg-FlowLogs",
```

Only the display name gets changed as per https://github.com/Azure/azure-policy/blob/master/built-in-policies/README.md#preview-and-deprecated-policies

**Review by jtracey93 — CHANGES_REQUESTED** [review]
sorry @JamJarchitect one more thing 👍

File: src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Nsg-FlowLogs.json (hunk 4 above)
Comment (JamJarchitect) [reply src/resources/Microsoft.Authorization/policyDefinitions/Deploy-Nsg-FlowLogs.json]: Didn't do it in `Deploy-Nsg-FlowLogs-to-LA.json`!!! I must've got "deprecation" happy 🤷‍♂️ amended and pushed

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**